skylib/kustomize/kustomize.bzl

```diff
@@ -479,49 +480,61 @@ def _kubectl_impl(ctx):
     if "{" in ctx.attr.user:
         user_arg = stamp(ctx, user_arg, files, ctx.label.name + ".user-name", True)
 
+    kubectl = "kubectl"
+    if ctx.executable.kubectl:
+        kubectl = "./" + ctx.executable.kubectl.short_path
+        files.append(ctx.executable.kubectl)
+        transitive = depset(transitive = [transitive, ctx.attr.kubectl.default_runfiles.files])
+
     kubectl_command_arg = ctx.attr.command
     kubectl_command_arg = ctx.expand_make_variables("kubectl_command", kubectl_command_arg, {})
 
-    statements = ""
-    transitive = None
+    statements = []
 
     if ctx.attr.push:
         trans_img_pushes = depset(transitive = [obj[KustomizeInfo].image_pushes for obj in ctx.attr.srcs]).to_list()
-        statements += "\n".join([
-            "echo pushing {}/{}:{}".format(exe[PushInfo].registry, exe[PushInfo].repository, exe[PushInfo].tag)
+        statements += [
+            """echo pushing {}/{}:{}""".format(exe[PushInfo].registry, exe[PushInfo].repository, exe[PushInfo].tag)
             for exe in trans_img_pushes
-        ]) + "\n"
-        statements += "\n".join([
-            "async \"${RUNFILES}/%s\"" % _get_runfile_path(ctx, exe.files_to_run.executable)
+        ]
+        statements += [
+            """async \"${{RUNFILES}}/{}\"""".format(_get_runfile_path(ctx, exe.files_to_run.executable))
```

Comment:
Is there any particular reason why you ha switched to a multiline string here (and couple other places)? Most of rules code originated from Google uses `"double quoted strings"` in situations like those.